app/routes/team/index.js

```diff
@@ -35,6 +35,12 @@ export default AuthenticatedRoute.extend(styleBody, CurrentUserSettings, Paginat
                 });
             }
 
+            // we need to load the roles into ember cache
+            // invites return role_id only and we do not offer a /role/:id endpoint
+            this.get('store').query('role', {permissions: 'assign'}).then((roles) => {
```

Comment:
This should be `modelPromises.roles = this.get('store').query('role', ...` and the promise body should just `return roles;`.

This is because the `model()` hook will pause for the returned promises to be resolved before continuing so when returning `RSVP.hash` you end up with a hash containing the values returned from the promises so the controller and everything else relying on the `model` object doesn't need to worry about things being async.